include/seastar/core/timer-set.hh

```diff
@@ -151,15 +160,28 @@ public:
      * Postconditions:
      *  - timer is no longer in the active set.
      *  - this object will no longer hold any references to this timer.
+     *
+     * get_next_timeout() will return correct next time point after removing
+     * timer with earliest timeout.
      */
     void remove(Timer& timer) noexcept
```

Comment:
What is the incorrect behavior you're observing exactly, what do you mean by "unable to set system timer callback". E.g. new timers never fire?

Reply:
The case is cased by timer self cancel in periodic timer callback. Sleep time for reactor were calculated badlly (in past). That caused reactor never come from sleep.

Reply:
The confusing comment has been removed.